Update 'PYTHON_VERSION_VENV' to 3.10 and add python3.10 to tox configs

sumincho22 opened the pull request:
CLOUDBLD-3117

Signed-off-by: Sumin Cho <sucho@redhat.com>

Since Cachito services run on Python 3.10, the unit tests should also cover this version as well. 

Should we get rid of Python 3.8 from the unit tests now? Three runs of unit testing seems time-consuming.

Also, we will need to add Python 3.10 to GitHub Actions.

# Maintainers will complete the following section

- [ ] Commit messages are descriptive enough
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] OpenAPI schema is updated (if applicable)
- [ ] DB schema change has corresponding DB migration (if applicable)
- [ ] README updated (if worker configuration changed, or if applicable)

lkolacek commented:
IMO we can drop Py 3.8 form unit tests. Other thoughts?

taylormadore commented:
I think we'll also need to update the python-version in a few places in the gh workflows config: https://github.com/containerbuildsystem/cachito/blob/28a6afc130df5296945898b7d8578e5bf2157ac0/.github/workflows/gating.yaml#L17

sumincho22 commented:
> IMO we can drop Py 3.8 form unit tests. Other thoughts?

- Removed Python 3.8.
- Made 3.9 the base version for unit and integration tests.
- Fixed GitHub Actions to run only 3.9 and 3.10 on unit tests

ejegrova reviewed (commented):
Please update also:
https://github.com/containerbuildsystem/cachito/blob/28a6afc130df5296945898b7d8578e5bf2157ac0/setup.py#L41
https://github.com/containerbuildsystem/cachito/blob/28a6afc130df5296945898b7d8578e5bf2157ac0/setup.py#L48

lkolacek commented:
And also:
```
tox.ini:35:basepython = python3.8
```

sumincho22 commented:
When I change `target-version` to `py39`, I also have to update `black` to a version greater than 20.8b1.

ejegrova commented:
> When I change `target-version` to `py39`, I also have to update `black` to a version greater than 20.8b1.
> 
> Should I change `black` to the newest version? If we do this, their formatting check also changes, so we would need to reformat the code base for it to pass.

It's better to have the latest version. If it works, please do these changes.

sumincho22 commented:
Updated `black` to version 22.6.0 (latest). Applied new formatting requirements.

taylormadore commented on .github/workflows/gating.yaml:
```diff
@@ -129,7 +129,7 @@ jobs:
       uses: actions/setup-python@v2
       with:
-        python-version: 3.8
+        python-version: 3.9
```
If python 3.10 is our target environment, should we use that for the integration tests?

sumincho22 replied:
But if the tests run successfully on python 3.9, then wouldn't python 3.10 pass as well? For this reason, I thought that the lower bound version would be better since we are testing both 3.9 and 3.10.

ejegrova replied:
Exactly, integration tests run only on python 3.9 now. Switch it to python 3.10 please.

lkolacek replied:
+1 to use latest greatest!

sumincho22 replied:
Gotcha, switched to python 3.10

sumincho22 commented on tox.ini:
```diff
@@ -32,12 +32,11 @@ commands_post =
 [testenv:black]
 # a bug in python3.9 is causing occasional failures: https://bugs.python.org/issue43498,
```
Yep I will remove it. I will a similar comment in `pyproject.toml`

sumincho22 commented:
@lkolacek @taylormadore @ejegrova This should be ready for approval and merge :)

ejegrova reviewed (approved):
looks great!
